### PR TITLE
Empty cue fix

### PR DESCRIPTION
**Pull request by LRahlff**
Fixes the error when a cue is empty.
Its still needed to have a channel and at least one cue

### CODE REVIEW

**Review by Doralitze — CHANGES_REQUESTED** [review]
Could you please also add a test case for the cue handling with empty ones? maybe just copy the content of the show file and make sure that the filter can actually handle it. Other than that it looks fine to me. Compiling works and I'll perform an in depth manual test tomorrow.

File: test/cue_filter_test.cpp [hunk 1]
```diff
@@ -16,1446 +16,1160 @@ struct Iomanager_Init {
...
+            if (tester_time == 1000) {
+                update_key_values[tester_time].push_back(std::tuple("run_mode", "play"));
+            }
+            test_values[tester_time].eight_bit_frames.push_back(0);
+        }
 
-    std::map <std::string, std::string> initial_parameters;
+        test_cue_function(time_s, test_values, channel_names, configuration, update_key_values);
+    }
+    BOOST_AUTO_TEST_CASE(betweenemptycues) {
```
Comment (LRahlff) [inline test/cue_filter_test.cpp:185]: added testcase

**Review by Doralitze — APPROVED** [review]
LGTM
